src/node.js

```diff
@@ -388,6 +396,53 @@
       nextTickQueue.push(obj);
       tickInfo[kLength]++;
     }
+
+    // Internal callbacks that must be run when there is no more micro-tasks
+    // scheduled for this turn.
```

Comment:
@vkurchatkin why? This is for scheduling an action to run after all micro-tasks in the queue have run. How would a queue of promises help us here?